### DISCUSSION TITLE
clickhouse-keeper  async_replication

### DISCUSSION THREAD

**Discussion by hert525** [ClickHouse/ClickHouse#65138] [Q&A]
Which version of ClickHouse Keeper introduced the async_replication parameter? Can I directly upgrade from version 22.6.4.35 without any risk?

**Comment by yiyang-shao** [comment]
According to the [documentation](https://clickhouse.com/docs/en/guides/sre/keeper/clickhouse-keeper#configuration).
>note
async_replication is disabled by default to avoid breaking backwards compatibility. If you have all your Keeper instances in a cluster running a version supporting async_replication (v23.9+), we recommend enabling it because it can improve performance without any downsides.

**Reply by hert525** [reply]
Thank you. I would like to know if my current ClickHouse version, which is 22.6.4.35, can be directly upgraded to version 23.9 or later. Are there any risks of data corruption or incompatibility?

**Reply by yiyang-shao** [reply]
I guess it will be a question regarding the generic process of upgrading clickhouse safely. I would not say that the answer is a simple yes or no.

I recommend the following practices to minimize risk and also checkout our [upgrade guide](https://clickhouse.com/docs/en/operations/update).
* **Check Release Notes:** Thoroughly review the release notes and changelogs for each intermediate version between your current version and the target version. This helps identify any breaking changes, deprecated features, and new configurations that may affect your upgrade.
* **Backup Data:** Before performing any upgrade, ensure you have a complete backup of your data. This allows you to restore your data if anything goes wrong during the upgrade process.
* **Test Upgrade in Staging:** Perform the upgrade in a staging environment first to identify and resolve any issues before applying the changes to your production environment.
* **Monitor and Verify:** After the upgrade, closely monitor your ClickHouse instance for any anomalies and verify that all functionalities are working as expected.
